include/triton/Conversion/MLIRTypes.h

```diff
@@ -10,20 +10,20 @@ namespace triton {
 namespace type {
 
 // Integer types
-Type i32Ty(MLIRContext *ctx) { return IntegerType::get(ctx, 32); }
-Type i8Ty(MLIRContext *ctx) { return IntegerType::get(ctx, 8); }
-Type u32Ty(MLIRContext *ctx) {
+static Type i32Ty(MLIRContext *ctx) { return IntegerType::get(ctx, 32); }
```

Comment:
I don't think `static` is needed here in the header file, it will duplicate the function in every cpp include.
ref https://stackoverflow.com/questions/780730/c-c-static-function-in-header-file-what-does-it-mean

Reply:
Without static, if two cpps includes this file, there will be a linkage error due to duplicate symbols. Moreover, functions like `static bool isFloat` in the same file use static, I'm just making it aligned to avoid future linkage errors.

`inline` is just an indicator and it will not force the compiler to do. If the function is long or recurisve, the compiler can choose not to inline even marked with an `inline`. So in this case, if two cpps includes this "inline" function, there is also a linkage error. For this consideration, I normally do not write inline to prevent linkage errors. O2/O3 is enough to automatically do that.

Reply:
The motivation for me to do that is I try to separate more functions into .h/.cpp (I didn't commit, just try). But the linker showed errors here, so I changed it to avoid future errors.